lib/active_merchant/billing/gateways/sage_pay.rb

```diff
@@ -195,14 +215,11 @@ def add_customer_data(post, options)
       end
 
       def add_optional_data(post, options)
-        add_pair(post, :GiftAidPayment, options[:gift_aid_payment]) unless options[:gift_aid_payment].blank?
```

Comment:
I had a look and `add_pair` handles this because it will only add a pair if the value isn't blank or if the option is required:
```
def add_pair(post, key, value, options = {})
          post[key] = value if !value.blank? || options[:required]
end 
```